src/test/unit_tests/test_s3select.js

```diff
@@ -0,0 +1,216 @@
+/* Copyright (C) 2023 NooBaa */
+'use strict';
+
+const mocha = require('mocha');
+const assert = require('assert');
+const s3select_utils = require('../../util/s3select');
+const stream = require('stream');
+const nb_native = require('../../util/nb_native')();
+const { Transform } = require('readable-stream');
+const stream_utils = require('../../util/stream_utils');
+
+const csv_small_str =
+`a,b,c,d,e,
+0,414,339,155,67,
+1,60,741,755,698,
+2,572,53,375,241,
+3,966,233,168,371,
+4,858,55,964,901,
+5,630,524,728,489,
+6,355,260,504,32,
+7,636,736,662,50,
+8,75,169,470,487,
+9,910,225,185,834,
+`;
+
+const json_str =
+`[
+{"doublef": 3.3, "intf" : 3, "stringf": "i am a string"},
+{"doublef": 3.3, "intf" : 123, "stringf": "i am a string"}
+]
+`;
+
+/*
+Handles streaming an input (csv/json) and getting the sql result as output.
+The stream in src/util/s3select.js also encodes the result into chunks according to AWS-defined format,
+which is not suitable here.
+The chunk parsing is tested by ceph's s3select.
+(If we do want to reuse rc/util/s3select.js, it's possible to factor out the encoding, eg add a parameter or add a stream
+to handle encoding specifically.)
+*/
+class S3SelectStream extends Transform {
+
+    constructor(context) {
+        super({});
+        this.s3select = new (nb_native.S3Select)(context);
+    }
+
+    async _transform(chunk, encoding, cb) {
+        //console.log("got chunk ", chunk.length);
+        let select = await this.s3select.write(chunk);
+        if (select) {
+            this.push(select.select);
+        }
+        return cb();
+    }
+
+    async _flush(cb) {
+        //console.log("in flush");
+        let select = await this.s3select.flush();
+        if (select) {
+            this.push(select.select);
+        }
+        return cb();
+    }
+}
+
+class CollectChunks extends stream.Writable {
+
+    chunks = [];
+
+    _write(chunk, enc, next) {
+        //console.log("got chunk");
+        this.chunks.push(chunk);
+        next();
+    }
+
+}
+
+/* DummyTransform is used a middle stream between
+input and output streams, like S3SelectStream.
+It is used to check that output stream is properly
+closed when input stream has errors*/
```

Comment:
Would you please move `/*` and `*/` to separate lines? (to match the style of other multi-line comments in this file - for example, 185-189 and 202-206)?